internal/request/graphql/schema/generate.go

```diff
@@ -1085,11 +1097,27 @@ func (g *Generator) GenerateMutationInputForGQLType(obj *gql.Object) ([]*gql.Fie
 
 func (g *Generator) genTypeFieldsEnum(obj *gql.Object) *gql.Enum {
 	enumFieldsCfg := gql.EnumConfig{
-		Name:   genTypeName(obj, "Fields"),
+		Name:   genTypeName(obj, typeFieldEnumSuffix),
+		Values: gql.EnumValueConfigMap{},
+	}
+
+	for f, field := range obj.Fields() {
+		enumFieldsCfg.Values[field.Name] = &gql.EnumValueConfig{Value: f}
+	}
+
+	return gql.NewEnum(enumFieldsCfg)
+}
+
+func (g *Generator) genUserExplicitTypeFieldsEnum(obj *gql.Object) *gql.Enum {
```

Comment:
praise: I like how you managed to do this, will hopefully be re-used in the future :)